dts/bindings/clock/st,stm32g0-hsi-clock.yaml

```diff
@@ -0,0 +1,30 @@
+# Copyright (c) 2022 STMicroelectronics
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  STM32 HSI Clock node description for STM32G0 devices
+  On STM32G0, HSI is a 16MHz fixed clock.
+
+  It also produces a HSISYS secondary clk which can be used as system clock
+  source. In that case, a HSI divisor (ranges from 1 to 128) can be applied:
+  SYSCLK = HSI16 / HSI DIV
+
+compatible: "st,stm32g0-hsi-clock"
+
+include: [fixed-clock.yaml]
+
+properties:
+    hsi-div:
+      type: int
+      required: true
+      description: |
+        HSI clock divider. Configures the output HSI clock frequency (HSISYS)
+      enum:
+        - 1   # hsi_clk = 16MHz
+        - 2   # hsi_clk = 8MHz
+        - 4   # hsi_clk = 4MHz
+        - 8   # hsi_clk = 2MHz
+        - 16  # hsi_clk = 1MHz
+        - 32  # hsi_clk = 0.5MHz
+        - 64  # hsi_clk = 0.25MHz
+        - 128 # hsi_clk = 0.125MHz
```

Comment:
Please move these useful comments explaining the values to the description instead of here.

That way they will show up in the bindings index: https://docs.zephyrproject.org/latest/build/dts/api/bindings.html

Reply:
thank you for your comment i will make the changes.